sklearn/model_selection/_validation.py

```diff
@@ -492,6 +493,43 @@ def _fit_and_score(estimator, X, y, scorer, train, test, verbose,
     return ret
 
 
+def _score_single_metric(estimator, X_test, y_test, scorer):
+    """Copmute the score of an estimator given a test set and a single metric.
+    This function will also check whether the estimator has a classes attribute
+    and pass that into the labels attribute of the scorer if available.
+    """
+    # check if estimator has a classes_ attribute and scorer has labels attr:
+    est_has_classes = hasattr(estimator, "classes_")
+    if hasattr(scorer, "_score_func"):
+        scorer_has_labels = "labels" in signature(
+                                        scorer._score_func).parameters
+    else:
+        scorer_has_labels = False
+
+    if est_has_classes & scorer_has_labels:
+        labels = getattr(estimator, "classes_", None)
+        scorer._set_score_func_parameters(labels=labels)
```

Comment:
yeah I don't think this is the right way. Why not do this when we get a string in ``check_scoring``? I feel like that if we get a scorer, it's the user's responsibility. And we could make ``labels`` a required argument in the future.